nornir/plugins/inventory/netbox.py

```diff
@@ -143,10 +147,66 @@ def __init__(
             "NB_TOKEN", "0123456789abcdef0123456789abcdef01234567"
         )
 
+        # Load in user private key from environment var or file load
+        nb_private_key = nb_private_key or os.environ.get("NB_PRIVATE_KEY", None)
+
+        if os.environ.get('NB_PRIVATE_KEY') is not None:
+            nb_private_key = os.environ.get('NB_PRIVATE_KEY')
+        elif os.environ.get('NB_PRIVATE_KEY') is None and nb_private_key:
+            with open(nb_private_key, 'r') as f:
+                nb_private_key = f.read()
+
         session = requests.Session()
         session.headers.update({"Authorization": f"Token {nb_token}"})
         session.verify = ssl_verify
 
+        # Fetch all secrets from Netbox
+        # Since the api uses pagination we have to fetch until no next is provided
+        if nb_private_key:
+            # Get session key first in order to get decrypted secrets
+
+            url = f"{nb_url}/api/secrets/get-session-key/"
+            nb_session_key: str = None
+
+            headers = {
```

Comment:
not sure if we really need to set these headers?

Reply:
I thought they were needed in testing with postman, ended up not being required. I was having a hell of a time getting a consistent response for the nb API for getting a session key. It would consistently send back a Set-Cookie for the session-key in the browser, but not in postman. That is also why I had to grab the data response and set it as a cookie manually.

Anyway, on this item, I removed it and it seems to work fine.